### DISCUSSION TITLE
How to open asset selector from custom Bard extension?

### DISCUSSION THREAD

**Discussion by david-windsock** [statamic/cms#9254] [Q&A]
We are trying to add a custom button to Bard editor that need to select an asset from asset selector. Is there a way to call it from JavaScript and return the asset path?. Just like the _Image_ button does...

**Comment by duncanmcclean** [comment]
Hey 👋

Your best bet would be to source dive the Image button that's built into Bard and try to copy it to work for your use case. 

For example: here's where the Image button shows the asset selector and the asset editor:

https://github.com/statamic/cms/blob/4.x/resources/js/components/fieldtypes/bard/Image.vue#L47-L69

You probably want to copy it and wire up its required props and events.

**Comment by david-windsock** [comment]
Thanks, but I was looking for a pure JS solution, because there is no Vue component involved. 

Our goal is to add support for inline attributes. Just added a new button using the [TipTap Image extension](https://tiptap.dev/docs/editor/api/nodes/image) and prompt for an URL like the example provided. Replacing the `prompt()` call for anything similar simplifies the task.

If there's no way to do it using JS API, we will create a Vue component as you suggest...

**Reply by duncanmcclean** [reply]
I don't think there's any way to avoid using a Vue component to do it.

**Reply by david-windsock** [reply]
Thanks, will try with Vue component instead...
